packages/forms/src/UIForm/Widget/condition.js

```diff
@@ -1,18 +1,167 @@
 import get from 'lodash/get';
-import includes from 'lodash/includes';
+import findIndex from 'lodash/findIndex';
 
-function getValue(value, strategy) {
-	if (!strategy) {
+/**
+ * Tries to ensure the returned value is a number.
+ * It supports string and number for now.
+ */
+function stringOrNumberToNumber(value) {
+	if (typeof value === 'number') {
 		return value;
 	}
-	switch (strategy.toLowerCase()) {
-		case 'length':
-			if (value && value.length) {
-				return value.length;
+	if (typeof value === 'string') {
+		return Number(value);
+	}
+	const error = { error: 'the passed value is not a string or a number', value };
+	throw error;
+}
+
+/**
+ * Tries to make the returned value a string.
+ * It supports number and string as input.
+ * In case of an error it will call onError. By default it throws an exception.
+ */
+function stringOrNumberToString(
+	value,
+	onError = val => {
+		const error = { error: 'the passed value is not a string or a number', value: val };
+		throw error;
+	},
+) {
+	if (!value) {
+		return value;
+	}
+	if (typeof value === 'number') {
+		return JSON.stringify(value);
+	}
+	if (typeof value === 'string') {
+		return value.toString();
+	}
+	return onError(value);
+}
+
+/**
+ * "foo=bar;dummy=1" will return [{foo:"bar",dummy:"1"}]
+ */
+function parseParameters(parameterString) {
+	return parameterString
+		.split(';')
+		.map(it => {
+			const sep = it.indexOf('=');
+			if (sep > 0) {
```

Comment:
It's still a good case for RegExp
Example : `^(?:(\w+)=)?(.+)$`

Define the key `value` as default if group1 matching is not found

This way we have a consistent unique return too and we not deal with indexes